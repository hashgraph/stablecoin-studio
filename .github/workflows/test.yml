name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test