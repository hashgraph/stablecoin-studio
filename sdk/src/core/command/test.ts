/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
