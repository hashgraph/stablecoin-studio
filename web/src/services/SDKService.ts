import { HederaNetwork, HederaNetworkEnviroment, NetworkMode, SDK } from 'hedera-stable-coin-sdk';
import type { AppMetadata, InitializationData , AcknowledgeMessage} from 'hedera-stable-coin-sdk';

export enum HashConnectConnectionState {
	Connected = 'Connected',
	Disconnected = 'Disconnected',
	Paired = 'Paired',
	Connecting = 'Connecting',
}

const appMetadata: AppMetadata = {
	name: 'dApp Example',
	description: 'An example hedera dApp',
	icon: 'https://absolute.url/to/icon.png',
	url: '',
};

interface CashInRequest {
	proxyContractId: string;
	privateKey: string;
	accountId: string;
	tokenId: string;
	targetId: string;
	amount: number;
}
interface EventsSetter {
	onInit: () => void;
	onWalletExtensionFound: () => void;
	onWalletPaired: () => void;
	onWalletAcknowledgeMessageEvent: (msg: AcknowledgeMessage) => void;
}

export class SDKService {
	private static instance: SDK | undefined;

	constructor() {}
	public static async getInstance(events?: EventsSetter) {
		if (!SDKService.instance) {
			SDKService.instance = new SDK({
				network: new HederaNetwork(HederaNetworkEnviroment.TEST), // TODO: dynamic data
				mode: NetworkMode.HASHPACK,
				options: {
					appMetadata,
				},
			});

			const { onInit, onWalletExtensionFound, onWalletPaired, onWalletAcknowledgeMessageEvent } =
				events || {
					onInit: () => {},
					onWalletAcknowledgeMessageEvent: () => {},
					onWalletExtensionFound: () => {},
					onWalletPaired: () => {},
				};

			await SDKService.instance.init({ onInit });
			SDKService.instance.onWalletExtensionFound(onWalletExtensionFound);
			SDKService.instance.onWalletPaired(onWalletPaired);
			SDKService.instance.onWalletAcknowledgeMessageEvent(onWalletAcknowledgeMessageEvent);
		}

		return SDKService.instance;
	}

	public static isInit() {
		// @ts-ignore
		return !!this.instance?.networkAdapter?._provider;
	}

	public static connectWallet() {
		SDKService.getInstance().then((instance) => instance?.connectWallet());
	}

	public static async getAvailabilityExtension(): Promise<boolean> {
		return (await SDKService.getInstance())?.getAvailabilityExtension();
	}

	public static async getStatus(): Promise<HashConnectConnectionState | undefined> {
		return (await SDKService.getInstance())?.gethashConnectConectionStatus();
	}

	public static async getWalletData(): Promise<InitializationData> {
		return (await SDKService.getInstance()).getInitData();
	}

	public static async disconnectWallet(): Promise<void> {
		return (await SDKService.getInstance()).disconectHaspack();
	}

	public static async cashIn({
		proxyContractId,
		privateKey,
		accountId,
		tokenId,
		targetId,
		amount,
	}: CashInRequest) {
		return await SDKService.getInstance().then((instance) =>
			instance.cashIn({ proxyContractId, privateKey, accountId, tokenId, targetId, amount }),
		);
	}
}

export default SDKService;
