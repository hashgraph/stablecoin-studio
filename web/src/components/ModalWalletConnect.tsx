import {
	Button,
	HStack,
	Image,
	Link,
	Modal,
	ModalBody,
	ModalContent,
	ModalFooter,
	ModalHeader,
	ModalOverlay,
	Spinner,
	Text,
	VStack,
	useDisclosure,
} from '@chakra-ui/react';
import {
	GetFactoryProxyConfigRequest,
	SupportedWallets,
	Network,
} from '@hashgraph/stablecoin-npm-sdk';
import type { StableCoinListViewModel } from '@hashgraph/stablecoin-npm-sdk';
import type { FC, ReactNode } from 'react';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useDispatch, useSelector } from 'react-redux';
import BLADE_LOGO_PNG from '../assets/png/bladeLogo.png';
import HASHPACK_LOGO_PNG from '../assets/png/hashpackLogo.png';
import FIREBLOCKS_LOGO_PNG from '../assets/png/fireblocksLogo.png';
import DFNS_LOGO_PNG from '../assets/png/dfnsLogo.png';
import METAMASK_LOGO from '../assets/svg/MetaMask_Fox.svg';
import SDKService from '../services/SDKService';
import {
	AVAILABLE_WALLETS,
	SELECTED_MIRRORS,
	SELECTED_RPCS,
	walletActions,
} from '../store/slices/walletSlice';
import WARNING_ICON from '../assets/svg/warning.svg';
import ERROR_ICON from '../assets/svg/error.svg';
import { SelectController } from './Form/SelectController';
import { useForm } from 'react-hook-form';
import type { IMirrorRPCNode } from '../interfaces/IMirrorRPCNode';
import type {FireblocksFormValues} from "./Form/FireblocksFormModal";
import FireblocksFormModal from "./Form/FireblocksFormModal";

const ModalWalletConnect = () => {
	const { t } = useTranslation('global');
	const dispatch = useDispatch();

	const {
		isOpen: isWalletSelectOpen,
		onOpen: onWalletSelectOpen,
		onClose: onWalletSelectClose,
	} = useDisclosure({ defaultIsOpen: true });

	const {
		isOpen: isFireblocksFormOpen,
		onOpen: onFireblocksFormOpen,
		onClose: onFireblocksFormClose,
	} = useDisclosure();


	const styles = {
		providerStyle: {
			boxShadow: '0 0 12px 2px #E0E0E0',
			borderRadius: 10,
			p: 6,
			_hover: {
				cursor: 'pointer',
				boxShadow: '0 0 12px 6px #E0E0E0',
				transform: 'scale(1.05)',
			},
		},
	};

	const stylesNetworkOptions = {
		menuList: {
			maxH: '220px',
			overflowY: 'auto',
			bg: 'brand.white',
			boxShadow: 'down-black',
			p: 4,
		},
		wrapper: {
			border: '1px',
			borderColor: 'brand.black',
			borderRadius: '8px',
			height: 'initial',
		},
	};

	const [loading, setLoading] = useState<SupportedWallets | undefined>(undefined);
	// eslint-disable-next-line @typescript-eslint/no-unused-vars
	const [error, setError] = useState<any>();
	const [rejected, setRejected] = useState<boolean>(false);
	const [hashpackSelected, setHashpackSelected] = useState<boolean>(false);
	const [bladeSelected, setBladeSelected] = useState<boolean>(false);
	const availableWallets = useSelector(AVAILABLE_WALLETS);
	const selectedMirrors: IMirrorRPCNode[] = useSelector(SELECTED_MIRRORS);
	const selectedRPCs: IMirrorRPCNode[] = useSelector(SELECTED_RPCS);


	const { control, getValues } = useForm({
		mode: 'onChange',
	});

	const handleWalletConnect = async (wallet: SupportedWallets, network: string) => {
		if (loading) return;
		setLoading(wallet);
		dispatch(walletActions.setLastWallet(wallet));
		dispatch(walletActions.setNetwork(network));
		dispatch(walletActions.setSelectedStableCoin(undefined));
		dispatch(walletActions.setSelectedStableCoinProxyConfig(undefined));
		dispatch(walletActions.setSelectedNetworkFactoryProxyConfig(undefined));
		dispatch(walletActions.setIsProxyOwner(false));
		dispatch(walletActions.setIsPendingOwner(false));
		dispatch(walletActions.setIsAcceptOwner(false));

		try {
			let mirrorNode;
			if (selectedMirrors.length > 0) {
				const listMirrors = selectedMirrors.filter(
					(obj: IMirrorRPCNode) =>
						obj.Environment.toLocaleLowerCase() === network.toLocaleLowerCase(),
				);
				if (listMirrors) mirrorNode = listMirrors[0];
			}
			let rpcNode;
			if (selectedRPCs.length > 0) {
				const listRPCs = selectedRPCs.filter(
					(obj: IMirrorRPCNode) =>
						obj.Environment.toLocaleLowerCase() === network.toLocaleLowerCase(),
				);
				if (listRPCs) rpcNode = listRPCs[0];
			}

			const result = await SDKService.connectWallet(wallet, network, mirrorNode, rpcNode);

			const newselectedMirrors: IMirrorRPCNode[] = [];

			selectedMirrors.forEach((obj) => {
				newselectedMirrors.push(obj);
			});

			if (!mirrorNode) {
				newselectedMirrors.push(result[1] as IMirrorRPCNode);
			}

			dispatch(walletActions.setSelectedMirrors(newselectedMirrors));

			const newselectedRPCs: IMirrorRPCNode[] = [];

			selectedRPCs.forEach((obj) => {
				newselectedRPCs.push(obj);
			});

			if (!rpcNode) {
				newselectedRPCs.push(result[2] as IMirrorRPCNode);
			}

			dispatch(walletActions.setSelectedRPCs(newselectedRPCs));

			const factoryId = await Network.getFactoryAddress();

			if (factoryId) {
				const factoryProxyConfig: StableCoinListViewModel = await getFactoryProxyConfig(factoryId);
				dispatch(walletActions.setSelectedNetworkFactoryProxyConfig(factoryProxyConfig));
			}
			dispatch(walletActions.setIsFactoryProxyOwner(false));
			dispatch(walletActions.setIsFactoryPendingOwner(false));
			dispatch(walletActions.setIsFactoryAcceptOwner(false));
		} catch (error: any) {
			if ('errorCode' in error && error.errorCode === '40009') {
				setRejected(true);
			} else {
				setError(error.message);
			}
			setLoading(undefined);
		}
	};

	const getFactoryProxyConfig = async (factoryId: string): Promise<StableCoinListViewModel> => {
		const factoryProxyConfig: any = await Promise.race([
			SDKService.getFactoryProxyConfig(
				new GetFactoryProxyConfigRequest({
					factoryId,
				}),
			),
			new Promise((resolve, reject) => {
				setTimeout(() => {
					reject(new Error("Stablecoin details couldn't be obtained in a reasonable time."));
				}, 10000);
			}),
		]).catch((e) => {
			if (e.code === 'NETWORK_ERROR') {
				throw new Error('The RPC service is not working as expected');
			}
			throw e;
		});
		return factoryProxyConfig;
	};

	const handleConnectHashpackWallet = () => {
		setHashpackSelected(true);
	};

	const unHandleConnectHashpackWallet = () => {
		setHashpackSelected(false);
		setLoading(undefined);
	};

	const handleConnectHashpackWalletConfirmed = () => {
		const values = getValues();
		handleWalletConnect(SupportedWallets.HASHPACK, values.networkHashpack.value);
	};

	const networkOptions = [{ value: 'testnet', label: 'Testnet' }];
	if (
		process.env.REACT_APP_ONLY_TESTNET === undefined ||
		process.env.REACT_APP_ONLY_TESTNET === 'false'
	) {
		networkOptions.push({ value: 'mainnet', label: 'Mainnet' });
	}

	const handleConnectMetamaskWallet = () => {
		handleWalletConnect(SupportedWallets.METAMASK, '-');
	};

	const handleConnectFireblocks = () => {
		// handleCustodialWalletConnect(SupportedWallets.FIREBLOCKS, '-');
		onWalletSelectClose(); // Cierra el modal de selecciÃ³n de wallet
		onFireblocksFormOpen(); // Abre el modal del formulario
	};

	const handleFireblocksFormConfirm = (formData: FireblocksFormValues) => {
		console.log("Datos del formulario:", formData);

		onFireblocksFormClose();
		// handleWalletConnect(SupportedWallets.FIREBLOCKS, '-');

	};

	const handleConnectBladeWallet = () => {
		setBladeSelected(true);
	};

	const unHandleConnectBladeWallet = () => {
		setBladeSelected(false);
		setLoading(undefined);
	};

	const handleConnectBladeWalletConfirmed = () => {
		const values = getValues();
		handleWalletConnect(SupportedWallets.BLADE, values.networkBlade.value);
	};

	const PairingSpinner: FC<{ wallet: SupportedWallets; children?: ReactNode }> = ({
		wallet,
		children,
	}) => {
		return (
			<>
				{loading && loading === wallet && (
					<HStack w={20} justifyContent='center' alignItems={'center'} h='full'>
						<Spinner
							w={50}
							h={50}
							justifyContent='center'
							alignSelf={'center'}
							color={wallet === SupportedWallets.HASHPACK ? '#C6AEFA' : '#f39c12'}
							thickness='8px'
						/>
					</HStack>
				)}
				{(!loading || loading !== wallet) && children}
			</>
		);
	};

	const userAgent = navigator.userAgent;

	const isChrome = userAgent.indexOf('Chrome') !== -1;

	return (
		<>
			<Modal
				isOpen={true}
				onClose={onWalletSelectClose}
				size={'xl'}
				isCentered
				closeOnEsc={false}
				closeOnOverlayClick={false}
			>
				<ModalOverlay />
				<ModalContent data-testid='modal-action-content' p='50' maxW="960px">
					{!error && !rejected && !hashpackSelected && !bladeSelected && (
						<>
							<ModalHeader p='0' justifyContent='center'>
								<Text
									data-testid='title'
									fontSize='20px'
									fontWeight={700}
									textAlign='center'
									lineHeight='16px'
									color='brand.black'
								>
									{t('walletActions.selectWallet')}
								</Text>
							</ModalHeader>
							<ModalFooter p='0' justifyContent='center'>
								<HStack
									spacing={10}
									pt={8}
									w='full'
									justifyContent={'center'}
									alignItems={'stretch'}
								>
									{availableWallets.includes(SupportedWallets.HASHPACK) ? (
										<VStack
											data-testid='Hashpack'
											{...styles.providerStyle}
											shouldWrapChildren
											onClick={handleConnectHashpackWallet}
										>
											<PairingSpinner wallet={SupportedWallets.HASHPACK}>
												<Image src={HASHPACK_LOGO_PNG} w={20} />
												<Text>Hashpack</Text>
											</PairingSpinner>
										</VStack>
									) : (
										<VStack data-testid='Hashpack' {...styles.providerStyle}>
											<Link
												href='https://www.hashpack.app/download'
												isExternal
												_hover={{ textDecoration: 'none' }}
											>
												<Image src={HASHPACK_LOGO_PNG} w={20} />
												<Text>Hashpack</Text>
											</Link>
										</VStack>
									)}
									{availableWallets.includes(SupportedWallets.METAMASK) ? (
										<VStack
											data-testid='Metamask'
											{...styles.providerStyle}
											shouldWrapChildren
											onClick={handleConnectMetamaskWallet}
										>
											<PairingSpinner wallet={SupportedWallets.METAMASK}>
												<Image src={METAMASK_LOGO} w={20} />
												<Text textAlign='center'>Metamask</Text>
											</PairingSpinner>
										</VStack>
									) : (
										<VStack data-testid='Metamask' {...styles.providerStyle}>
											<Link
												href='https://metamask.io/download/'
												isExternal
												_hover={{ textDecoration: 'none' }}
											>
												<Image src={METAMASK_LOGO} w={20} />
												<Text textAlign='center'>Metamask</Text>
											</Link>
										</VStack>
									)}
									{isChrome ? (
										availableWallets.includes(SupportedWallets.BLADE) ? (
											<VStack
												data-testid='Blade'
												{...styles.providerStyle}
												shouldWrapChildren
												onClick={handleConnectBladeWallet}
											>
												<PairingSpinner wallet={SupportedWallets.BLADE}>
													<Image src={BLADE_LOGO_PNG} w={20} />
													<Text textAlign='center'>Blade</Text>
												</PairingSpinner>
											</VStack>
										) : (
											<VStack data-testid='Blade' {...styles.providerStyle}>
												<Link
													href='https://bladewallet.io/'
													isExternal
													_hover={{ textDecoration: 'none' }}
												>
													<Image src={BLADE_LOGO_PNG} w={20} />
													<Text textAlign='center'>Blade</Text>
												</Link>
											</VStack>
										)
									) : (
										<></>
									)}
									<VStack
										data-testid='Fireblocks'
										{...styles.providerStyle}
										shouldWrapChildren
										onClick={handleConnectFireblocks}
									>
										<PairingSpinner wallet={SupportedWallets.FIREBLOCKS}>
											<Image src={FIREBLOCKS_LOGO_PNG} w={20} />
											<Text textAlign='center'>Fireblocks</Text>
										</PairingSpinner>
									</VStack>
									<VStack
										data-testid='Dfns'
										{...styles.providerStyle}
										shouldWrapChildren
										onClick={handleConnectFireblocks}
									>
										<PairingSpinner wallet={SupportedWallets.DFNS}>
											<Image src={DFNS_LOGO_PNG} w={20} />
											<Text textAlign='center'>DFNS</Text>
										</PairingSpinner>
									</VStack>
								</HStack>
							</ModalFooter>
						</>
					)}
					{hashpackSelected && (
						<>
							<ModalHeader p='0' justifyContent='center'>
								<Text
									fontSize='20px'
									fontWeight={700}
									textAlign='center'
									lineHeight='16px'
									color='brand.black'
								>
									{t('walletActions.selectWallet')}
								</Text>
							</ModalHeader>
							<ModalFooter alignSelf='center' pt='24px' pb='0'>
								<VStack>
									<SelectController
										control={control}
										isRequired
										name='networkHashpack'
										defaultValue='0'
										options={networkOptions}
										addonLeft={true}
										overrideStyles={stylesNetworkOptions}
										variant='unstyled'
									/>
									<HStack>
										<Button
											data-testid='modal-notification-button-Hashpack'
											onClick={unHandleConnectHashpackWallet}
											variant='secondary'
										>
											{t('common.cancel')}
										</Button>
										<Button
											data-testid='modal-notification-button-Hashpack'
											onClick={handleConnectHashpackWalletConfirmed}
											variant='primary'
										>
											{t('common.accept')}
										</Button>
									</HStack>
								</VStack>
							</ModalFooter>
						</>
					)}
					{bladeSelected && (
						<>
							<ModalHeader p='0' justifyContent='center'>
								<Text
									fontSize='20px'
									fontWeight={700}
									textAlign='center'
									lineHeight='16px'
									color='brand.black'
								>
									{t('walletActions.selectWallet')}
								</Text>
							</ModalHeader>
							<ModalFooter alignSelf='center' pt='24px' pb='0'>
								<VStack>
									<SelectController
										control={control}
										isRequired
										name='networkBlade'
										defaultValue='0'
										options={networkOptions}
										addonLeft={true}
										overrideStyles={stylesNetworkOptions}
										variant='unstyled'
									/>
									<HStack>
										<Button
											data-testid='modal-notification-button-Blade'
											onClick={unHandleConnectBladeWallet}
											variant='secondary'
										>
											{t('common.cancel')}
										</Button>
										<Button
											data-testid='modal-notification-button-Blade'
											onClick={handleConnectBladeWalletConfirmed}
											variant='primary'
										>
											{t('common.accept')}
										</Button>
									</HStack>
								</VStack>
							</ModalFooter>
						</>
					)}
					{(error || rejected) && (
						<>
							<ModalHeader alignSelf='center' p='0'>
								<Image
									data-testid='modal-notification-icon'
									src={error ? ERROR_ICON : WARNING_ICON}
									width='54px'
									height='54px'
								/>
							</ModalHeader>
							<ModalBody textAlign='center' pt='14px'>
								<Text
									data-testid='modal-notification-title'
									fontSize='14px'
									fontWeight={700}
									lineHeight='16px'
									color='brand.black'
								>
									{error ?? t('pairing.rejected')}
								</Text>
							</ModalBody>
							<ModalFooter alignSelf='center' pt='24px' pb='0'>
								<Button
									data-testid='modal-notification-button'
									onClick={() => {
										dispatch(walletActions.clearData());
										setError(undefined);
										setRejected(false);
									}}
									variant='primary'
								>
									{t('common.goBack')}
								</Button>
							</ModalFooter>
						</>
					)}
				</ModalContent>
			</Modal>
			<FireblocksFormModal
				isOpen={isFireblocksFormOpen}
				onClose={onFireblocksFormClose}
				onConfirm={handleFireblocksFormConfirm}
			/>
		</>
	);
};

export default ModalWalletConnect;
