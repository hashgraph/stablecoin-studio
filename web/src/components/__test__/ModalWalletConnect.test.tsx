import { render } from '../../test/index';
import ModalWalletConnect from '../ModalWalletConnect';
import translations from '../../translations/en/global.json';
import configureMockStore from 'redux-mock-store';
import { SupportedWallets } from '@hashgraph/stablecoin-npm-sdk';
import userEvent from '@testing-library/user-event';
import SDKService from '../../services/SDKService';
import type { IMirrorRPCNode } from '../../interfaces/IMirrorRPCNode';

const mockStore = configureMockStore();

describe(`<${ModalWalletConnect.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<ModalWalletConnect />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should has title', () => {
		const component = render(<ModalWalletConnect />);

		const title = component.getByTestId('title');

		expect(title).toHaveTextContent(translations.walletActions.selectWallet);
	});

	test('should has wallets and is clicked', async () => {
		const rpcNode: IMirrorRPCNode = {
			name: '::name::',
			BASE_URL: '::url::',
			API_KEY: '::key::',
			Environment: '::env::',
			isInConfig: true,
			HEADER: '::header::',
		};

		const store = mockStore({
			wallet: {
				foundWallets: [SupportedWallets.HASHPACK, SupportedWallets.METAMASK],
				selectedMirrors: rpcNode,
				selectedRPCs: rpcNode,
			},
		});
		const component = render(<ModalWalletConnect />, store);

		const metamask = component.getByTestId('Metamask');
		await userEvent.click(metamask);

		expect(SDKService.connectWallet).toHaveBeenCalledWith(
			'Metamask',
			expect.any(String),
			undefined,
			undefined,
		);
	});
});
