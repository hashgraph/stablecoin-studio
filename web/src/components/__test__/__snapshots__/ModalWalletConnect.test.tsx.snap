// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalWalletConnect /> should render correctly 1`] = `
<DocumentFragment>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
